admin/services/attendance.service.js

```diff
@@ -64,6 +64,19 @@ const attendanceService = {
    */
   getAttendanceCodes: async () => {
     return attendanceCodeDataService.sqlFindAttendanceCodes()
+  },
+
+  /**
+   * Check if pupil has valid attendance
+   * @param pupilId
+   * @param pinEnv
+   * @return {boolean}
+   */
+  hasValidAttendance: async (pupilId, pinEnv) => {
+    const pupilAttendance = await pupilAttendanceDataService.findOneByPupilId(pupilId)
+    return pinEnv === 'live'
+      ? pupilAttendance && pupilAttendance.id
```

Comment:
This returns either the `id` or the `attendanceCode_id` rather than the bool suggested in the jsdoc won't it?

Reply:
I've added some more comments in the jsdoc to explain what it does (plus fixed that bool thing).